libraries/common/binstream.effekt

```diff
@@ -0,0 +1,349 @@
+import option
```

Comment:
Naming 🚲 🏠 
1. Should this be a top-level module like `binstream` or a nested one like `stream/binary` / `stream/bin`...?
2. Is it a _binary_ stream or a _bit_ stream?